packages/flutter/test/material/theme_test.dart

```diff
@@ -47,6 +47,68 @@ void main() {
     expect(Theme.of(tester.element(find.text('menuItem'))).brightness, equals(Brightness.dark));
   });
 
+  group('Theme.brightnessOf', () {
+    testWidgets('return correct brightness when just media query is given', (
+      WidgetTester tester,
+    ) async {
+      await tester.pumpWidget(
+        const MediaQuery(
+          data: MediaQueryData(platformBrightness: Brightness.dark),
+          child: SizedBox(),
+        ),
+      );
+
+      expect(Theme.brightnessOf(tester.element(find.byType(SizedBox))), equals(Brightness.dark));
+    });
+
+    testWidgets('return correct brightness with overriding theme brightness over media query', (
+      WidgetTester tester,
+    ) async {
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(platformBrightness: Brightness.dark),
+          child: Theme(data: ThemeData(brightness: Brightness.light), child: const SizedBox()),
+        ),
+      );
+
+      expect(Theme.brightnessOf(tester.element(find.byType(SizedBox))), equals(Brightness.light));
+    });
+  });
+
+  group('Theme.maybeBrightnessOf', () {
```

Comment:
To test the value when no MediaQuery or ThemeData is present, you can use a test like this:

```
    testWidgets('returns null when no theme or media query is present', (
      WidgetTester tester,
    ) async {
      await tester.pumpWidget(
        const SizedBox(),
      );

      expect(Theme.maybeBrightnessOf(tester.element(find.byType(SizedBox))), isNull);
    });
```

Reply:
@navaronbracke I don't think this is going to return null, because internally its also using same `platformBrightnessOf` from MediaQuery.

Reply:
But it will return null though?

Theme.maybeBrightnessOf() looks at the Theme first, which is null here, so it then falls back to using

`MediaQuery.maybePlatformBrightnessOf(context)` which is documented to return null if there is no MediaQuery:

`Returns MediaQueryData.platformBrightness for the nearest MediaQuery ancestor or null, if no such ancestor exists.`

Reply:
```dart
testWidgets('returns null when no theme or media query is present', (
      WidgetTester tester,
    ) async {
      await tester.pumpWidget(const SizedBox());

      expect(
        Theme.maybeBrightnessOf(tester.element(find.byType(SizedBox))),
        equals(Brightness.light),
      );
    });
```

This is surprising, because this test is passing for me.

Reply:
May be I am missing something
```dart
      expect(find.byType(MediaQuery), findsOneWidget);
```
is passing test for same above test.

Reply:
Can use use `wrapWithView: false` when pumping the Widget? Maybe the default View includes a MediaQuery.

See https://api.flutter.dev/flutter/flutter_test/WidgetTester/pumpWidget.html

Otherwise I don't know what would add the default MediaQuery

Reply:
Aha: https://api.flutter.dev/flutter/widgets/View-class.html

```
[RawView](https://api.flutter.dev/flutter/widgets/RawView-class.html), the workhorse that [View](https://api.flutter.dev/flutter/widgets/View-class.html) uses to create the render tree, but without the [MediaQuery](https://api.flutter.dev/flutter/widgets/MediaQuery-class.html) and [FocusScope](https://api.flutter.dev/flutter/widgets/FocusScope-class.html) that [View](https://api.flutter.dev/flutter/widgets/View-class.html) adds.
```